### PR TITLE
Add static check

### PR DESCRIPTION
**Pull request by pankajastro**
Add a GitHub action job to run static checks and lint for the project.
- Add pre-commit hooks (pre-commit-hooks, ruff, black)
- Expose hatch command to run it in CI as well as local
Example: ```hatch run tests.py3.12-2.10:static-check```

closes: https://github.com/astronomer/dag-factory/issues/196

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+name: test
+
+on:
+  push: # Run on pushes to the default branch
+    branches: [main, pre-commit]
+  pull_request_target: # Also run on pull requests originated from forks
+    branches: [main, pre-commit]
```
Comment (tatiana) [inline .github/workflows/test.yml:7]:
```suggestion
    branches: [main]
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+name: test
+
+on:
+  push: # Run on pushes to the default branch
+    branches: [main, pre-commit]
```
Comment (tatiana) [inline .github/workflows/test.yml:5]:
```suggestion
    branches: [main]
```

**Review by tatiana — COMMENTED** [review]
It's a great addition, @pankajastro. This change will help us maintain the code quality! Thank you!

**Review by tatiana — APPROVED** [review]

**Review by pankajkoti — COMMENTED** [review]
Nice change, thanks @pankajastro 👏🏽

**Pull request merged**